### DISCUSSION TITLE
Issues with Portainer and Proxmox and OpenID connect

### DISCUSSION THREAD

**Discussion by thimplicity** [authelia/authelia#5531] [Q&A] [answered]
Hi, I am running authelia on docker and have successfully connected my services with HAproxy and lldap as the backend. Portainer and lldap wok fine. Now I am trying to get OpenID up and running and I started with Portainer and Proxmox. Both are not working, but in this post, I will focus on Portainer, as I think that the resolution for that probably also resolves the Proxmox problem. There is no logging in docker for that Portainer login, only for Proxmox.

My authelia config portion for identity providers:

`identity_providers:
  oidc:
    hmac_secret: ...
    issuer_private_key: |
      -----BEGIN RSA PRIVATE KEY-----
      ...
      -----END RSA PRIVATE KEY-----
    access_token_lifespan: 1h
    authorize_code_lifespan: 1m
    id_token_lifespan: 1h
    refresh_token_lifespan: 90m
    enable_client_debug_messages: false
    enforce_pkce: public_clients_only
    cors:
      endpoints:
        - authorization
        - token
        - revocation
        - introspection
    clients:
     - id: portainer
        description: Portainer
        secret: '...'
        public: false
        authorization_policy: one_factor
        redirect_uris:
          - https://portainer....
        audience: []
        scopes:
          - openid
          - profile
          - groups
          - email
        userinfo_signing_algorithm: none`

Portainer config screenshot:

<img width="835" alt="image" src="https://github.com/authelia/authelia/assets/42503549/42d39dc6-e789-4ff0-93a5-d45b810801c6">

Portainer "Use OAuth" error message:
`{"error":"invalid_request","error_description":"The request is missing a required parameter, includes an invalid parameter value, includes a parameter more than once, or is otherwise malformed. The 'redirect_uri' parameter does not match any of the OAuth 2.0 Client's pre-registered redirect urls."}`

Thanks for some guidance!

**Comment by james-d-elliott** [comment, marked as answer]
The redirect URI that portainer is requesting doesn't match the one in the Authelia config using simple string comparison semantics.

**Comment by james-d-elliott** [comment]
An off topic point, please stop posting a discussion and an issue for the same thing, issues are for things that are clearly a bug or for feature requests, discussions are for everything else. 

If you post an issue it must use the template and all required fields in the template must be included and completed per the template instructions.

**Reply by thimplicity** [reply]
Hi James,
thanks for reminding me. I would love to do that. For some reason, when I click on "New Issue" it redirects me to creating a discussion. This repo is the only one where that happens to me, so the only way I found to create an issue was to create a discussion and move it into an issue. See screenshot below for the screen I see when clicking on new issue:

<img width="1237" alt="image" src="https://github.com/authelia/authelia/assets/42503549/284b2adf-f832-4d40-a27a-f5c20b8acc21">

Maybe I am doing something wrong though

**Reply by james-d-elliott** [reply]
Thanks for letting me know, looks like GitHub changed their YAML parser somehow. However nothing you've reported is an actual issue. It's support.
